File: 2597. The Number of Beautiful Subsets/Solution.py
class Solution:
    def beautifulSubsets(self, nums: List[int], k: int) -> int:
        return sum([self.backtrack(nums, k, i, set()) for i in range(len(nums))])
    def backtrack(self, nums, k, i, seen):
        if i == len(nums): return 0
        total = 1
        seen.add(nums[i])
        for j in range(i + 1, len(nums)):
            if nums[j] - k in seen or nums[j] + k in seen: continue
            total += self.backtrack(nums, k, j, seen)
        seen.discard(nums[i])
        return total

File: 2597. The Number of Beautiful Subsets/Solution.java
class Solution {
    public int beautifulSubsets(int[] nums, int k) {
        int res = 0;
        for (int i = 0; i < nums.length; i++) {
            res += backtrack(nums, k, i, new HashSet<>());
        }
        return res;
    }
    public int backtrack(int[] nums, int k, int i, Set<Integer> curr) {
        if (i == nums.length) return 0;
        int total = 1;
        curr.add(nums[i]);
        for (int j = i + 1; j < nums.length; j++) {
            if (curr.contains(nums[j] - k) || curr.contains(k + nums[j])) continue;
            total += backtrack(nums, k, j, curr);
        }
        curr.remove(nums[i]);
        return total;
    }
}